pkg/controller/jobframework/reconciler_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobframework
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	kubeflow "github.com/kubeflow/mpi-operator/pkg/apis/kubeflow/v2beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	testingjob "sigs.k8s.io/kueue/pkg/util/testingjobs/job"
+	testingmpijob "sigs.k8s.io/kueue/pkg/util/testingjobs/mpijob"
+)
+
+func TestIsParentJobManaged(t *testing.T) {
+	parentJobName := "test-job-parent"
+	childJobName := "test-job-child"
+	jobNamespace := "default"
+	cases := map[string]struct {
+		parentJob client.Object
+		job       client.Object
+		want      bool
+		wantErr   string
+	}{
+		"child job doesn't have ownerReference": {
+			parentJob: testingmpijob.MakeMPIJob(parentJobName, jobNamespace).
+				UID(parentJobName).
+				Obj(),
+			job: testingjob.MakeJob(childJobName, jobNamespace).
+				Obj(),
+			wantErr: "owner isn't set even though \"kueue.x-k8s.io/parent-workload\" annotation is set",
+		},
+		"child job has ownerReference with unknown workload owner": {
+			parentJob: testingjob.MakeJob(parentJobName, jobNamespace).
+				UID(parentJobName).
+				Obj(),
+			job: testingjob.MakeJob(childJobName, jobNamespace).
+				OwnerReference(parentJobName, batchv1.SchemeGroupVersion.WithKind("Job")).
+				Obj(),
+			wantErr: fmt.Sprintf("workload owner %v is unknown", &metav1.OwnerReference{
+				APIVersion: batchv1.SchemeGroupVersion.String(),
+				Kind:       "Job",
+				Name:       parentJobName,
+				UID:        types.UID(parentJobName),
+				Controller: pointer.Bool(true),
+			}),
+		},
+		"child job has ownerReference with known non-existing workload owner": {
+			job: testingjob.MakeJob(childJobName, jobNamespace).
+				OwnerReference(parentJobName, kubeflow.SchemeGroupVersionKind).
+				Obj(),
+			wantErr: "mpijobs.kubeflow.org \"test-job-parent\" not found",
```

Comment:
are these errors typed? Do they implement `Is`? If so, we should prefer comparing actual errors than strings.

Reply:
> are these errors typed?

No, that isn't errors typed. This error will return from https://github.com/kubernetes-sigs/kueue/blob/c44773e711779a6aaec49011a0f249b4fc61a49c/pkg/controller/jobframework/reconciler.go#L262C1-L263.

Reply:
done.